src/actions/index.js

```diff
@@ -1,15 +1,56 @@
-export const addCard = (name, email) => ({
+const errorAfterFetch = err => ({
+  type: 'ERROR',
+  err,
+});
+
+const fetchCardsSucces = cards => ({
+  type: 'FETCH_CARDS',
+  cards,
+});
+
+const addCardSucces = card => ({
   type: 'ADD_CARD',
-  name,
-  email,
-  id: Date.now(),
+  name: card.name,
+  email: card.email,
+  id: card.id,
 });
 
-export const deleteCard = id => ({
+const deleteCardSucces = id => ({
   type: 'DELETE_CARD',
   id,
 });
 
+const url = 'http://localhost:3004/cards';
+
+export const fetchCards = () => dispatch => fetch(url)
+  .then(res => res.json())
+  .then(cards => dispatch(fetchCardsSucces(cards)))
+  .catch(err => dispatch(errorAfterFetch(err)));
+
+
+export const addCard = (name, email) => dispatch => fetch(url,
+  {
+    method: 'post',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ name, email }),
+  })
+  .then(res => res.json())
+  .then(card => dispatch(addCardSucces(card)))
+  .catch(err => dispatch(errorAfterFetch(err)));
+
+export const deleteCard = id => dispatch => fetch(url + id,
```

Comment:
There is something wrong with card removal. When I click remove it's removed from the list, but after reload the cards are all back visible. (Even the ones added by myself).
Also DELETE requests return '404' response code

Reply:
I found a bug, I forget to add "/" on url, in later commits, I fix that bug.

Reply:
Yup, saw that later, that you fixed it in following PR's